draft-ietf-anima-brski-prm.md

```diff
@@ -1103,589 +1166,583 @@ pledgeenrollrequesttrigger = {
-|        |    |   Agent    |    | Registrar |    |   CA   |    |      |
+| Pledge |    | Registrar- |    |  Domain   |    |  Key   |    | MASA |
+|        |    |   Agent    |    | Registrar |    | Infra. |    |      |
 +--------+    +------------+    +-----------+    +--------+    +------+
  |                  |                 |                 |   Internet |
  ~                  ~                 ~                 ~            ~
-(4) Supply PER to Registrar (including backend interaction)
+(4) Supply PER to Registrar (including Key Infrastructure interaction)
  ~                  ~                 ~                 ~            ~
  |                  |                 |                 |            |
  |                  |<----(mTLS)----->|                 |            |
  |                  |-------PER------>|                 |            |
- |                  |                 |<-----mTLS------>|            |
- |                  |                 |-------RER------>|            |
- |                  |                 |<--Enroll-Resp---|            |
+ |                  |                 |----[Request]--->|            |
+ |                  |                 |<--[Certificate]-|            |
  |                  |<--Enroll-Resp---|                 |            |
  |                  |                 |                 |            |
  ~                  ~                 ~                 ~            ~
 ~~~~
 {: #exchangesfig_uc2_4 title="Enroll exchange" artwork-align="center"}
 
-In case the TLS connection to the registrar is already closed, the Registrar-Agent opens a new TLS connection with the registrar as stated in {{pvr}}.
+As a second step of the interaction with the domain registrar, the Registrar-Agent SHALL supply the PER artifact(s) to the registrar via HTTP-over-TLS POST to the registrar endpoint at `/.well-known/brski/requestenroll`.
+The request body MUST contain one previously acquired PER artifact as defined in {{per-artifact}}.
+In the request header, the Content-Type field MUST be set to `application/jose+json` and the Accept field SHOULD be set to `application/jose+json`.
 
-### Request Artifact: Pledge Enroll-Request (PER)
+Note that this is different from the EST {{RFC7030}} endpoint used in BRSKI, as the PER artifact is signature-wrapped.
+Hence, upon receiving a PER artifact, the registrar MUST verify that
 
-As specified in {{tper}} deviating from BRSKI the PER is not a raw PKCS#10.
-As the Registrar-Agent is involved in the exchange, the PKCS#10 is wrapped in a JWS object by the pledge and signed with pledge's IDevID to ensure proof-of-identity as outlined in {{per_example}}.
+* the PER was signed signed with the private key corresponding to the pledge EE certificate, which is contained in the JWS Protected Header of the PER.
+* the pledge identified by its EE certificate is accepted to join the domain after successful validation of the corresponding PVR.
```

Comment:
EE cert to also address re-enroll case.
Do we need more text or clarify "accepted to join", as re-enroll is a different decision (already joined)?

Reply:
For the re-enrollment case the part "is accepted to join the domain after successful validation of the corresponding PVR." would be omitted, as the pledge already has been accepted in the first run in the domain. 
Proposal to enhance the bullet point to 

* the pledge identified by its EE certificate is accepted to join the domain after successful validation of the corresponding PVR in case of the initial enrollment using the pledge's IDevID certificate. For the re-enrollment the pledge uses its still valid LDevID certificate, which identifies him as already accepted component in the domain.

Reply:
Moved it to the text about re-enrollment at the end to keep the thoughts together:

> Note that while BRSKI-PRM targets the initial enrollment, re-enrollment may be supported in a similar way with the exception that the current pledge LDevID certificate is used instead of the IDevID certificate to sign the PER artifact (see also {{tper}}).
Hence, there is no verification whether the pledge is accepted to join the domain, as the still valid LDevID certificate identifies the pledge as already accepted component of the domain.

Reply:
Perfect!

Reply:
Done